### DISCUSSION TITLE
[v8] Is there a Typescript regression?

### DISCUSSION THREAD

**Discussion by kosciak9** [TanStack/table#4265] [Q&A] [answered]
Hello,

Updating to v8.5.6 fails our type checks - apparently our way of defining columns is wrong:

```ts
const columnHelper = createColumnHelper<Transaction>();

const columns = [
  columnHelper.accessor("date", {
    header: () => "Data",
  }),
  columnHelper.accessor("title", {
    header: () => "Tytuł",
  }),
  columnHelper.accessor(
    (row) => `${row.amount ?? 0} ${row.currency?.symbol ?? ""}`,
    {
      id: "amountDisplay",
      header: () => "Kwota",
    }
  ),
```

We are not using `ColumnDef` as Typescript properly inferred types from current revision of code and everything was working properly. Is this something that we should change? Or is there unwanted regression?

**Comment by tannerlinsley** [comment, marked as answer]
Working on a fix

Tanner Linsley

**Reply by kosciak9** [reply]
Amazing, then we're gonna hold updating for now
